doc/examples/joystick/lib/dynamic_joystick.dart

```diff
@@ -0,0 +1,69 @@
+import 'dart:math';
+import 'dart:ui';
+
+import 'package:flame/game/base_game.dart';
+import 'package:flame/gestures.dart';
+import 'package:flame/joystick.dart';
+import 'package:flame/sprite.dart';
+import 'package:flutter/gestures.dart';
+
+import 'player.dart';
+
+class DynamicJoystick extends Joystick {
+  DynamicJoystick() {
+    // The joystick is invisible by default and only shown when the player touches the screen
+    visible = false;
+    fixedPosition = false;
+    // Setting the visuals of the base and knob
+    base.sprite = Sprite('joystick_base.png');
+    knob.sprite = Sprite('joystick_knob.png');
+  }
+
+  @override
+  void resize(Size size) {
+    // The size of the joystick will be 20% of the screen's width
+    final side = size.width * 0.2;
+    knob.width = knob.height = side;
+    base.width = base.height = side;
+    movementRadius = side * sqrt2 / 2; // half the hypotenuse of the square
+  }
+}
+
+class DynamicJoystickGame extends BaseGame with PanDetector {
+  final player = Player();
+  final joystick = DynamicJoystick();
+
+  DynamicJoystickGame() {
+    add(player);
+    add(joystick);
+  }
+
+  void _startInput(Offset pos) => joystick.showAt(pos);
+
+  void _updateInput(Offset pos) {
```

Comment:
I think we should make this transparent to the developer, as it can be kind boilerplate to use.

I think we can probably use a similar solution used on Tapables components, where the listener of that component was direct hooked on the widgets builder.

@luanpotter @spydon @renancaraujo your thoughts?

Reply:
Just a clarification, the StaticJoystick and DynamicJoystick are just examples on how someone can use the Joystick class. The Joystick class itself is not abstract, but I think it's a good practice to extend it to setup everything and use the resize method to adjust it according to the screen.

The Joystick class doesn't handle input automatically, just provides usefull methods that can be fed input by the developer, like I did in this example.

Reply:
> The Joystick class doesn't handle input automatically, just provides usefull methods that can be fed input by the developer, like I did in this example.

indeed it can't handle, but my point is that I think that it should, as one of the main reasons of the components to exist on Flame, is to make it easy and transparent to use.

Of course that "auto handling" could be a thing to be added on the future and we can go like that atm, but still is something that I want to discuss about.

Reply:
Oh ok got it. It's just that there are a lot of ways (or more then one at least) that developers might want to configure the input, so I thought this way gives them more freedom.